server/routerlicious/packages/services/src/throttlerHelper.ts

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    IThrottler,
+    IThrottlerHelper,
+    IThrottlerResponse,
+    ThrottlerRequestType,
+    ThrottlingError,
+    ILogger,
+} from "@fluidframework/server-services-core";
+
+/**
+ * A lenient implementation of IThrottlerHelper that prioritizes low latency over strict throttling.
+ */
+export class ThrottlerHelper implements IThrottlerHelper {
+    // TODO: Should these cache values expire? Use node-cache package perhaps?
```

Comment:
Hesitant to add an external dependency like `node-cache` to an otherwise external-dependency-free library, especially since `node-cache` isn't a dependency in the rest of R11S. Any suggestions for avoiding a memory leak here? While implementing a `ICache` with simple expiry is not difficult, I do not want to unnecessarily reinvent the wheel.

One partial solution that comes to mind is using `delete` for various resets, rather than resetting a value. For example, `this.throttlerResponseCache[key] = { throttleStatus: false, ... }` would simply become `delete this.throttlerResponseCache[key]`. However, this would not solve for lastThrottleUpdateAtMap, because the reset value there is important, so some type of expiration logic would be useful.

Reply:
I think its okay to use a lightweight caching library here. May be something like an lru cache with limited size?